From 4b609315b31391d85b6391106d53fa4b3b940508 Mon Sep 17 00:00:00 2001
From: gaulthier gain <gaulthier.gain@uliege.be>
Date: Fri, 7 Aug 2020 15:29:24 +0000
Subject: [PATCH] Add macro to avoid conflicts with arc4random

This patch adds the `EVENT__HAVE_ARC4RANDOM_BUF` macro which is
defined in the `config.h` of the libevent repository in order
to avoid conflicts with libc (musl/newlib) since `arc4random_buf`
is already defined.

Signed-off-by: gaulthier gain <gaulthier.gain@uliege.be>
---
 arc4random.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arc4random.c b/arc4random.c
index a2338e6..7b29923 100644
--- a/arc4random.c
+++ b/arc4random.c
@@ -494,6 +494,7 @@ arc4random(void)
 }
 #endif
 
+#ifndef EVENT__HAVE_ARC4RANDOM_BUF
 ARC4RANDOM_EXPORT void
 arc4random_buf(void *buf_, size_t n)
 {
@@ -507,6 +508,7 @@ arc4random_buf(void *buf_, size_t n)
 	}
 	ARC4_UNLOCK_();
 }
+#endif
 
 #ifndef ARC4RANDOM_NOUNIFORM
 /*
-- 
2.17.1

